### PR TITLE
GCM zero size IV

### PR DESCRIPTION
**Pull request by karel-m**
see #273

### CODE REVIEW

File: src/encauth/gcm/gcm_done.c [hunk 1]
```diff
@@ -41,6 +41,8 @@ int gcm_done(gcm_state *gcm,
    }
 
    if (gcm->mode == LTC_GCM_MODE_IV) {
+      /* IV length must be > 0 */
```
Comment (sjaeckel) [inline src/encauth/gcm/gcm_done.c]: this check shouldn't be required as `gcm_add_aad()` already checks for it

File: src/encauth/gcm/gcm_process.c [hunk 2]
```diff
@@ -53,6 +53,8 @@ int gcm_process(gcm_state *gcm,
    }
 
    if (gcm->mode == LTC_GCM_MODE_IV) {
+      /* IV length must be > 0 */
```
Comment (sjaeckel) [inline src/encauth/gcm/gcm_process.c]: same

**Comment by sjaeckel** [comment]
good find! :+1:

**Comment by karel-m** [comment]
I see what you mean, but all those extra (duplicate) `if (gcm->mode == LTC_GCM_MODE_IV) { .. }` where added in #246 - GCM allow skipping gcm_add_aad and gcm_process.

After #246 it is legal to completely skip `gcm_add_aad` and/or `gcm_process` therefore zero IV size has to be checked IMO in all 3 places (gcm_add_aad, gcm_process, gcm_done).

**Comment by sjaeckel** [comment]
okay, but these two checks are then either on the wrong place or they can be removed.

These checks are both in the state `gcm->mode == LTC_GCM_MODE_IV` immediately before a call to `gcm_add_aad()`. `gcm_add_aad()` has in the state `gcm->mode == LTC_GCM_MODE_IV` as first line the same check, therefore these two checks can be removed. Or am I wrong?

**Comment by karel-m** [comment]
Oh, you are right.

**Pull request merged**